middleware/introspection/instrumentedhandler.go

```diff
@@ -1,4 +1,4 @@
-package instrospection
+package introspection
```

Comment:
This matches the directory name…

Reply:
That's a sign nothing is actually using anything defined in this file. That's alarming.